### PR TITLE
feat: support `delete_after` for ephemeral messages

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

The API recently started supporting `DELETE /webhooks/.../@original`; `inter.delete_original_response()` already works fine, however `inter.response.send_message(delete_after=)` didn't.

Not sure if this should be considered a bugfix instead of a feature.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/interactions/base.py [hunk 1]
```diff
@@ -858,6 +861,9 @@ async def send_message(
             before deleting the message we just sent. If the deletion fails,
             then it is silently ignored.
 
+            .. versionchanged:: 2.7
+                Added support for ephemeral messages.
```
Comment (onerandomusername) [inline disnake/interactions/base.py:865]: note that in the case of an ephemeral message, they can only be deleted up to a maximum of 15 minutes.

**Review by onerandomusername — COMMENTED** [review]
This is a feature, because it supports a new api feature.

There may be a bug with InteractionMessage not falling back properly in the case of ephemerals, I'm not sure. (this also might be mentioned as a seperate issue already)

File: disnake/interactions/base.py (hunk 1 above)
Comment (shiftinv) [reply disnake/interactions/base.py]: https://github.com/DisnakeDev/disnake/pull/816/commits/d9324ce869847f5ad94967c06aebf5f27a6bb96c

File: disnake/interactions/base.py [hunk 2]
```diff
@@ -858,6 +864,12 @@ async def send_message(
             before deleting the message we just sent. If the deletion fails,
             then it is silently ignored.
 
+            Can be up to 15 minutes, after which the interaction token expires
```
Comment (onerandomusername) [inline disnake/interactions/base.py:867]: up to 15 minutes from the creation of the interaction***

Comment (shiftinv) [reply disnake/interactions/base.py]: https://github.com/DisnakeDev/disnake/pull/816/commits/23ed979d5aaf1f6d3a19ae8d552e3fc5a3108446 something like this?

Comment (onerandomusername) [reply disnake/interactions/base.py]: yeah, i think so

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**